source/identifier.go

```diff
@@ -191,6 +194,36 @@ func FromLLB(op *pb.Op_Source, platform *pb.Platform) (Identifier, error) {
 			}
 		}
 	}
+	if id, ok := id.(*OCIIdentifier); ok {
+		if platform != nil {
+			id.Platform = &ocispecs.Platform{
+				OS:           platform.OS,
+				Architecture: platform.Architecture,
+				Variant:      platform.Variant,
+				OSVersion:    platform.OSVersion,
+				OSFeatures:   platform.OSFeatures,
+			}
+		}
+		for k, v := range op.Source.Attrs {
+			switch k {
+			case pb.AttrLocalSessionID:
+				id.SessionID = v
+				if p := strings.SplitN(v, ":", 2); len(p) == 2 {
+					id.Name = p[0] + "-" + id.Name
+					id.SessionID = p[1]
+				}
+			case pb.AttrImageLayerLimit:
```

Comment:
Can this be passed from the client side atm? I think it might be simpler to only keep `ImageIdentifier` in here an add the session key to it. We still want properties from Image like layerlimit and recordtype.

Reply:
Honestly? No idea. I just took it from the `ImageIdentifier` section above. I think it would want to be consistent.

Reply:
This should be `AttrOCILayoutLayerLimit/AttrOCILayoutSessionID`